### PR TITLE
[react-native] Move out of ‘react’ into its own namespace and add a few more declarations.

### PR DESCRIPTION
**Pull request by alloy**
* Since RN 0.25, the RN specific APIs are no longer in the `react` module: https://github.com/facebook/react-native/releases/tag/v0.25.1
* Adds `findNodeHandle`: https://github.com/facebook/react-native/blob/v0.42.0/Libraries/Renderer/src/renderers/native/findNodeHandle.js#L50
* Adds `XMLHttpRequest` and `originalXMLHttpRequest` to `global` so that it can be rewired in dev mode and network requests show up in Chrome’s network pane: https://github.com/facebook/react-native/issues/934

I have verified that these changes work in our app.

### CODE REVIEW

**Comment by andy-ms** [comment]
Could you avoid using a namespace and just export things directly?
We actually have a lint rule for that, but `react-native` is currently not linted.

**Comment by alloy** [comment]
@andy-ms Done. I’ve also:

* volunteered myself as maintainer of this package
* made the linter green and enabled it (assuming adding the `tslint.json` file is the only requirement to enable it?)

File: react-native/index.d.ts [hunk 1]
```diff
@@ -18,6 +18,12 @@
 
 /// <reference types="react" />
 
+// Disabling because:
+// * Object -> object requires TypeScript 2.2
+// * The Function type seems to be a pretty good choice for most-all places in this file.
+//
+// tslint:disable:forbidden-types
```
Comment (alloy) [inline react-native/index.d.ts]: If anyone has any better ideas about this, please do tell.

File: react-native/tslint.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "../tslint.json",
+    "rules": {
+        // This package uses the Function type for good reason and `object` is TypeScript 2.2 only.
+        "forbidden-types": false
```
Comment (alloy) [inline react-native/tslint.json]: If anyone has any better ideas about this, please do tell.

Comment (alloy) [inline react-native/tslint.json]: If anyone has any better ideas about this, please do tell.

Comment (andy-ms) [inline react-native/tslint.json]: What's the reason you use the `Function` type?

Comment (andy-ms) [inline react-native/tslint.json]: Also, in the case of `Object`, does it really have to not be a primitive?

Comment (alloy) [inline react-native/tslint.json]: Regarding `Function`, there’s for instance this profile function that wraps any given function: https://github.com/facebook/react-native/blob/fbd1b0a6654d87b236a382d52575d637f87acf24/Libraries/Performance/Systrace.js#L183-L192

Comment (alloy) [inline react-native/tslint.json]: Good point re those `Object` cases, I’ll take another look to see if there are places that can be resolved by other more specific types.

Comment (andy-ms) [inline react-native/tslint.json]: In that case, the output should have an identical type as the input. So you can take a `<T extends Function>` and return a `T`.

**Comment by alloy** [comment]
I was able to get rid of all the `Function` types, but getting rid of `Object` proved less ideal, so I’m thinking that it makes more sense to wait till we can use TS 2.2 and move to `object`. The question then becomes, when can we use TS 2.2?

**Comment by andy-ms** [comment]
TS2.2 is available now, but since not everyone is on it yet, I don't think it's worth requiring it for this package just to fix a linter warning. So I would leave the disable in and fix it later.

**Comment by alloy** [comment]
Agreed 👍

**Comment by alloy** [comment]
Is there still something that needs to be addressed?

**Comment by alloy** [comment]
Rebased.

**Pull request merged**